RT_UPDATE/download_SRB.R

```diff
@@ -0,0 +1,177 @@
+#' If the specified location is not found in the WPP data, a default SRB value corresponding to a sex ratio of
+#' 105 male births per 100 female births (\code{1.0486}) is returned. If certain requested years are missing from
+#' the WPP dataset, the missing SRB values are imputed using the mean of available SRB values.
+#'
+#' @return
+#' A named numeric vector of SRB values for each year in \code{DatesOut}.
+#'
+#' @importFrom dplyr filter select
+#' @importFrom tidyr pivot_longer
+#' @importFrom tibble as_tibble
+#' @importFrom rlang .data
+#' @importFrom magrittr %>% 
+#'
+#' @examples
+#' \dontrun{
+#' # Example 1: Download SRB data for France
+#' srb_data <- download_SRB(location = "France",
+#'                          DatesOut = c(2000, 2005, 2010))
+#'
+#' # Example 2: Use user-supplied constant SRB value
+#' srb_data <- download_SRB(SRB = 1.05,
+#'                          location = "France",
+#'                          DatesOut = c(2000, 2005, 2010))
+#' }
+#'
+#' @export
+
+download_SRB <- function(SRB = NULL, 
+                         location, 
+                         DatesOut, 
+                         verbose = TRUE) {
+  
+  SRB_default <- round((1 - 0.4886) / 0.4886, 3)
+  
+  # Check DatesOut
+  if(length(DatesOut) < 1) {
+    
+    stop("DatesOut must contain at least one date.")
+    
+  }  
+  
+  # If SRB provided directly
+  # ------------------------------------------------------------------ #
+  # CASE 1: User-supplied SRB values (no download required)
+  # ------------------------------------------------------------------ #
+  if (!is.null(SRB)) {
+    
+    SRB <- setNames(rep(SRB, length.out = length(DatesOut)), DatesOut)
+    
+    return(SRB)
+    
+  }
+  
+  # ------------------------------------------------------------------ #
+  # CASE 2: Retrieve SRB from latest installed WPP package
+  # ------------------------------------------------------------------ #
+  if(is.null(SRB)) { 
+  
+  installed_wpp <- grep("^wpp\\d{4}$", rownames(installed.packages()), value = TRUE)
+  
+  if(length(installed_wpp) == 0) stop("No WPP package installed.")
+  
+  latest_wpp <- sort(installed_wpp, decreasing = TRUE)[1]
+  
+  if(latest_wpp < 2022) {
+    
+    stop("No single years SRB is availabe in wpp versions earlier that wpp2022.
+           Please update the wpp package.")
+    
+  }
+  
+  # Load SRB data from the selected WPP package
+  data("sexRatio1", package = latest_wpp)
+  
+  # Determine country code (numeric or character location input)
+  if(is.numeric(location)) {
+    
+    location_code <- location
+    
+  } else {
+    
+    location_code <- sexRatio1 %>%
+      as_tibble() %>%
+      filter(.data$name %in% location) %>%
+      select("country_code") %>%
+      unique() %>%
+      as.numeric()
+  }
+  
+  # Handle missing or invalid locations gracefully
+  if(is.na(location_code)) {
+    
+    if(verbose) {  
+      
+      cat(location, "not available in wpp. Using default SRB:", SRB_default, "\n")
```

Comment:
always use message() instead of cat(). This is a forthcoming change throughout DemoTools..